### ISSUE TITLE
["Request"] Disambiguate `unit` function in Applicative/Functor

### ISSUE THREAD

**Issue by aballano** [arrow-kt/arrow#2001]
We have a duplicated function name that performs 2 completely different things

unit for Applicative
`fun unit(): Kind<F, Unit> = just(Unit)`

and unit for Functor
`fun <A> Kind<F, A>.unit(): Kind<F, Unit> = map { Unit }`

The problem with this can be easily spotted when having an extension function over an interface extending, for example, MonadError, where you wanna start calling unit(), as both functions can be resolved, there's no guarantee of which one will be called, causing possible bugs. Here unit could either mean `just(Unit)` or `this.map { Unit }`:

```
fun <A> Kind<F, A>.something(...): Kind<F, A> =
    unit().doSomethingElse()
```

I'd suggest removing/renaming the unit() from Functor as the name is not indicating a map transformation whatsoever. Maybe simply `mapUnit`? cc @arrow-kt/maintainers

**Comment by pakoito** [comment]
I'm okay with dealing away with the functor one

**Comment by 1Jajen1** [comment]
I'd say the opposite, the `Applicative` one seems much less useful to me.
The functor one is to remove/void any content of a datatype and mainly used to match type signatures.
The applicative version is different because it is only ever used as a filler, it is a type with no effects, not return value so it cannot be used otherwise. It is mainly used when functions expect arguments of type `f Unit` and you intend to perform no effects. But where would you want to do this? If the argument is that uninteresting to the function that you can fill it with `just(Unit)` it might indicate that the function does not need it (maybe add an overload). If the argument is important to the function (like the release part of a bracket) then you should not use `just(Unit)` in the first place (why bother with bracketing if you don't do anything on release?)

Imo the functor is more useful.

**Comment by pakoito** [comment]
Should we rename the functor one to `void`? Is it a legal function name :D

**Comment by 1Jajen1** [comment]
> Should we rename the functor one to `void`? Is it a legal function name :D

Should be in kotlin land right? `void` is a good idea. That is what haskell land uses and it accurately describes what it is doing (unlike `unit` as @aballano mentioned it does not imply `map` as the actual action). :+1:

**Comment by pakoito** [comment]
`devnull`

**Comment by 1Jajen1** [comment]
> `devnull()`

Perfect! :clap: :clap: :clap: :clap: :clap:

**Comment by nomisRev** [comment]
> I'd say the opposite, the Applicative one seems much less useful to me.

Yes, I use both all the time actually :D And I'd be sad to see one of the two go :p

> Should we rename the functor one to void? Is it a legal function name :D
Yes, that sounds good to me too 👍

Let's settle on `void` and change the description of the ticket. Concerns or ideas still welcome!
